### PR TITLE
feat(@clayui/drop-down): adds section API support to drop-down items

### LINKED ISSUE #2811: Dropdown Menu: Checkbox and Radio hover interaction styles inherited
## What are the steps to reproduce?

1. Go to [Clay Storybook](https://storybook.clayui.com/)
2. Open ClayDropdown menu item
3. Select Checkbox or Radio 
4. Click "Click me" button

## Current behavior

checkbox-hover

## What is the expected result?
Dropdown item hover or selected interaction styles mustn't be inherited to check boxes and radio buttons

## Environment

| Tech  | Version |
| ----- | ------- |
| Clay  | v3.0  |

### PR DESCRIPTION
**Pull request by matuzalemsteles**
Fixes #2811

### CODE REVIEW

File: packages/clay-drop-down/src/Item.tsx [hunk 1]
```diff
@@ -28,6 +28,13 @@ interface IProps
 
 	innerRef?: React.Ref<any>;
 
+	/**
+	 * Flag to indicate that the element is a session, are for cases when
+	 * Checkbox and Radio do not need the styles when hovering over the
+	 * drop-down item.
+	 */
+	isSection?: boolean;
```
Comment (matuzalemsteles) [inline packages/clay-drop-down/src/Item.tsx]: I'm not really sure about that name, I thought about maybe just leaving `section`. Suggestions?

Comment (bryceosterhaus) [reply packages/clay-drop-down/src/Item.tsx]: Yeah I think just `section` is a better fit for this, unless we want to create a new "Section" component for dropdown, but I think just a flag would be enough.

File: packages/clay-drop-down/src/Item.tsx [hunk 2]
```diff
@@ -28,6 +28,13 @@ interface IProps
 
 	innerRef?: React.Ref<any>;
 
+	/**
+	 * Flag to indicate that the element is a session, are for cases when
+	 * Checkbox and Radio do not need the styles when hovering over the
+	 * drop-down item.
+	 */
```
Comment (bryceosterhaus) [inline packages/clay-drop-down/src/Item.tsx]:
```suggestion
	/**
	 * Flag to indicate that the element is a section. This is used for cases like
	 * Checkbox and Radio where they do not need hover styles.
	 */
```

File: packages/clay-drop-down/src/Item.tsx (hunk 1 above)
Comment (matuzalemsteles) [reply packages/clay-drop-down/src/Item.tsx]: Hmm, I think a new component for "Section" makes sense, but in order not to duplicate it we would have to have an ItemBase to reuse the logic. What do you think? I think I like to have a new component for that even though what will only change is the className 😅

Comment (bryceosterhaus) [reply packages/clay-drop-down/src/Item.tsx]: @matuzalemsteles new component is fine with me. Any idea if we need to support the icon for a section? If not, we wont need to duplicate too much

Comment (matuzalemsteles) [reply packages/clay-drop-down/src/Item.tsx]: @bryceosterhaus I don't know but we can go without the icon support for now.

**Comment by matuzalemsteles** [comment]
@bryceosterhaus updated.

**Pull request merged**